hazelcast/src/main/java/com/hazelcast/config/QueueConfig.java

```diff
@@ -323,6 +327,57 @@ public QueueConfig setMergePolicyConfig(MergePolicyConfig mergePolicyConfig) {
         return this;
     }
 
+    /**
+     * Check if underlying implementation is a {@code PriorityQueue}. Otherwise it is a FIFO queue
+     *
+     * @return {@code true} if priority queue has been configured, {@code false} otherwise
+     */
+    public boolean isPriorityQueue() {
+        return comparatorClassName != null;
+    }
+
+    /**
+     * Returns the class name of the configured {@link Comparator} implementation.
+     *
+     * @return the class name of the configured {@link Comparator} implementation
+     */
+    public String getComparatorClassName() {
+        return comparatorClassName;
+    }
+
+    /**
+     * Sets the class name of the configured {@link Comparator} implementation.
+     *
+     * @param comparatorClassName the class name of the
+     *                            configured {@link Comparator} implementation
+     * @return this QueueConfig instance
+     */
+    public QueueConfig setComparatorClassName(String comparatorClassName) {
+        this.comparatorClassName = comparatorClassName;
+        return this;
+    }
+
+    /**
+     * Check if duplicates are allowed for this queue.
+     *
+     * @return {@code true} if duplicates are allowed, {@code false} otherwise
+     */
+    public boolean isDuplicateAllowed() {
```

Comment:
This seems to be very use case specific. I'm not sure we want to have this in the codebase.

Reply:
Hi Peter,
Thanks for your kind feedback so far.
You are right. This is really use case specific. I would have like to provide a solution without altering internal of Queue implementation. 
I am open for alternative solution.

Reply:
I don't think there is an alternative solution. But there is no queue implementation that has duplicate detection and I have never had the need for it. So I don't see a good reason why it should be in here; this is a general-purpose data-structure and we should prevent add use case-specific additions. It makes the code complex to maintain and use.

Reply:
What happens when a regular queue is created and duplicate allowed is set to false? Afaik there is no duplicate checking in the regular queue (would be undesirable due to complexity). So is there an exception if you set the false to false, but don't create a priority queue? Or is the setting ignored?

Reply:
Attribute duplicate-allowed in the config holds only for priority queue. This setting is completely ignored for regular queue.